apps/wing/src/analytics/collect.ts

```diff
@@ -0,0 +1,43 @@
+import { Command } from "commander";
+import { AnalyticEvent } from "./event";
+import { OSCollector } from "./collectors/os-collector";
+import { NodeCollector } from "./collectors/node-collector";
+import { CLICollector } from "./collectors/cli-collector";
+import { CICollector } from "./collectors/ci-collector";
+import { AnalyticsStorage } from "./storage";
+import { GitCollector } from "./collectors/git-collector";
+
+
+/**
+ * Collects analytics for a given command, stores it for later export
+ * 
+ * @param cmd The commander command to collect analytics for
+ * @returns string the file path of the stored analytic
+ */
+export async function collectCommandAnalytics(cmd: Command): Promise<string | undefined> {
+  const osCollector = new OSCollector();
+  const nodeCollector = new NodeCollector();
+  const ciCollector = new CICollector();
+  const gitCollector = new GitCollector();
+  const cliCollector = new CLICollector(cmd);
+
+  const eventName = `cli_${cmd.opts().target ?? ''}_${cmd.name()}`;
+
+  let event: AnalyticEvent = {
+    event: eventName.replace(/[^a-zA-Z_]/g, ""),
+    properties: {
+      cli: await cliCollector.collect(),
+      os: await osCollector.collect(),
+      node: await nodeCollector.collect(),
+      ci: await ciCollector.collect(),
+      anonymous_repo_id: (await gitCollector.collect())?.anonymous_repo_id
```

Comment:
Why is this not just?

```suggestion
      git: await gitCollector.collect()
```

Reply:
Current solution is ok (talked to Elad)